src/index.js

```diff
@@ -13,16 +13,118 @@ const { wrap } = require('@adobe/openwhisk-action-utils');
 const { logger } = require('@adobe/openwhisk-action-logger');
 const { wrap: status } = require('@adobe/helix-status');
 const { epsagon } = require('@adobe/helix-epsagon');
+const { TimeoutError } = require('@adobe/helix-fetch');
+const { MountConfig } = require('@adobe/helix-shared');
+const { fetchFSTab } = require('./github');
+
+const github = require('./github');
+const google = require('./google');
+const onedrive = require('./onedrive');
+
+const DEFAULT_FORWARD_HEADERS = [
+  'x-request-id',
+  'x-cdn-request-id',
+  'x-cdn-url',
+];
+
+const HANDLERS = [
+  github, google, onedrive,
+];
 
 /**
- * This is the main function
- * @param {string} name name of the person to greet
+ * Retrieves content from a content repository
+ * @param {string} owner github org or username
+ * @param {string} repo repository name
+ * @param {string} ref branch or tag name
+ * @param {string} path file path
  * @returns {object} a greeting
  */
-function main({ name = 'world' }) {
-  return {
-    body: `Hello, ${name}.`,
+async function main({
+  owner, repo, ref, path,
+  REPO_RAW_ROOT, HTTP_TIMEOUT, GITHUB_TOKEN,
+  __ow_headers: originalHeaders = {}, __ow_logger: log,
+}) {
+  if (!(owner && repo && ref && path)) {
+    return {
+      statusCode: 400,
+      body: 'owner, repo, ref, and path parameters are required',
+    };
+  }
+  const gitHubToken = GITHUB_TOKEN || originalHeaders['x-github-token'];
+  const githubRootPath = REPO_RAW_ROOT || 'https://raw.githubusercontent.com/';
+
+  const githubOptions = {
+    timeout: HTTP_TIMEOUT || 1000,
+    headers: DEFAULT_FORWARD_HEADERS.reduce((headers, header) => {
+      // copy whitelisted headers
+      if (originalHeaders[header.toLocaleLowerCase()]) {
+        // eslint-disable-next-line no-param-reassign
+        headers[header.toLocaleLowerCase()] = originalHeaders[header.toLowerCase()];
+      }
+      return headers;
+    }, {
+      // pass on authorization token
+      Authorization: gitHubToken ? `token ${gitHubToken}` : undefined,
+    }),
+  };
+
+  const externalOptions = {
+    timeout: HTTP_TIMEOUT ? HTTP_TIMEOUT * 10 : 10 * 1000,
```

Comment:
use separate timeout, as defined in:

https://github.com/adobe/helix-pipeline/blob/b8d87656c16e7494cd8432ae58c59945e847df03/src/schemas/secrets.schema.json#L73-L77

(also, default is 20s)